relay-metrics/src/aggregatorservice.rs

```diff
@@ -128,144 +94,250 @@ pub struct FlushBuckets {
+#[derive(Clone, Debug, Deserialize, Serialize)]
+#[serde(tag = "field", content = "value", rename_all = "lowercase")]
+pub enum Field {
+    /// Field that allows comparison to a metric or bucket's namespace.
+    Namespace(MetricNamespace),
+}
+
 enum AggregatorState {
     Running,
     ShuttingDown,
 }
 
-/// A service for aggregationg metric buckets.
-///
-/// Buckets are flushed to a receiver after their time window and a grace period have passed.
-/// Metrics with a recent timestamp are given a longer grace period than backdated metrics, which
-/// are flushed after a shorter debounce delay. See [`AggregatorServiceConfig`] for configuration options.
-///
-/// Internally, the aggregator maintains a continuous flush cycle every 100ms. It guarantees that
-/// all elapsed buckets belonging to the same [`ProjectKey`] are flushed together.
+/// Service that routes metrics & metric buckets to the appropriate aggregator.
 ///
-/// Receivers must implement a handler for the [`FlushBuckets`] message.
+/// Each aggregator gets its own configuration.
+/// Metrics are routed to the first aggregator which matches the configuration's [`Condition`].
+/// If no condition matches, the metric/bucket is routed to the `default_aggregator`.
 pub struct AggregatorService {
-    aggregator: aggregator::Aggregator,
-    state: AggregatorState,
-    receiver: Option<Recipient<FlushBuckets, NoResponse>>,
+    default_aggregator: aggregator::Aggregator,
+    secondary_aggregators: BTreeMap<MetricNamespace, aggregator::Aggregator>,
     max_flush_bytes: usize,
     max_total_bucket_bytes: Option<usize>,
     flush_partitions: Option<u64>,
+    state: AggregatorState,
+    receiver: Option<Recipient<FlushBuckets, NoResponse>>,
 }
 
 impl AggregatorService {
-    /// Create a new aggregator service and connect it to `receiver`.
-    ///
-    /// The aggregator will flush a list of buckets to the receiver in regular intervals based on
-    /// the given `config`.
+    /// Create a new aggregator service.
     pub fn new(
-        config: AggregatorServiceConfig,
+        aggregator_config: AggregatorServiceConfig,
+        secondary_aggregators: Vec<ScopedAggregatorConfig>,
         receiver: Option<Recipient<FlushBuckets, NoResponse>>,
     ) -> Self {
-        Self::named("default".to_owned(), config, receiver)
-    }
+        let max_total_bucket_bytes = aggregator_config.max_total_bucket_bytes;
+        let max_flush_bytes = aggregator_config.max_flush_bytes;
+        let flush_partitions = aggregator_config.flush_partitions;
+
+        if !secondary_aggregators.iter().all(|agg| {
+            agg.config.flush_partitions == flush_partitions
+                && agg.config.max_total_bucket_bytes == max_total_bucket_bytes
+                && agg.config.max_flush_bytes == max_flush_bytes
+        }) {
+            relay_log::error!("aggregatorserviceconfigs are inconsistent");
+        }
```

Comment:
These three settings used to be on each `AggregatorService`, now we've decided we only need them once for the router, but they are still available on each `AggregatorServiceConfig`, so we log an error if they're inconsistent, and simply use the ones from the default config. 

Perhaps we should have a separate `AggregatorServiceConfig` that doesn't include `AggregatorConfig`?

Reply:
Q: why not have a specific config for `AggregatorService`? Not sure if this requires two steps to keep compatibility with the config file.

Reply:
do you mean what i wrote in the last sentence?

>Perhaps we should have a separate AggregatorServiceConfig that doesn't include AggregatorConfig?

Mainly trying not to make this PR even bigger than it already is, and was unsure if there's some compatibility issues